MetaMask/metamask-extension
# Use legacy gas API for BSC

## Explanation

The legacy gas API is still useful for BSC, which is a network our APIs support that is not EIP-1559 compatible. The legacy gas API will now be used for BSC prior to using RPC methods as a fallback.

This brings extension closer in alignment with mobile, which also uses the legacy gas API for BSC.


The E2E network mock function has been updated to use a variable for the initial test network. This made it easier to write the e2e test for the BSC case.

## Manual Testing Steps

Check that gas estimates still work correctly on BSC (e.g. in Send flow and transaction confirmations).

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
